src/components/RegistrationList.tsx

```diff
@@ -1,28 +1,42 @@
 import CloseOutlinedIcon from '@mui/icons-material/CloseOutlined';
 import EditIcon from '@mui/icons-material/Edit';
+import StarIcon from '@mui/icons-material/Star';
+import StarOutlineIcon from '@mui/icons-material/StarOutline';
 import { Box, IconButton, List, ListItemText, Link as MuiLink, Tooltip, Typography } from '@mui/material';
+import { useMutation } from '@tanstack/react-query';
 import { useTranslation } from 'react-i18next';
-import { Link } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { Link, useLocation } from 'react-router-dom';
+import { updatePromotedPublications } from '../api/preferencesApi';
+import { setNotification } from '../redux/notificationSlice';
 import { Registration, RegistrationStatus } from '../types/registration.types';
 import { dataTestId } from '../utils/dataTestIds';
 import { displayDate } from '../utils/date-helpers';
 import { getTitleString } from '../utils/registration-helpers';
-import { getRegistrationLandingPagePath, getRegistrationWizardPath, getResearchProfilePath } from '../utils/urlPaths';
+import {
+  UrlPathTemplate,
+  getRegistrationLandingPagePath,
+  getRegistrationWizardPath,
+  getResearchProfilePath,
+} from '../utils/urlPaths';
 import { ContributorIndicators } from './ContributorIndicators';
 import { ErrorBoundary } from './ErrorBoundary';
 import { TruncatableTypography } from './TruncatableTypography';
 import { SearchListItem } from './styled/Wrappers';
-
 interface RegistrationListProps {
   registrations: Registration[];
   canEditRegistration?: boolean;
   onDeleteDraftRegistration?: (registration: Registration) => void;
+  promotedPublications?: string[] | undefined;
```

Comment:
(correct me if wrong :) )
```suggestion
  promotedPublications: string[] | undefined;
```